pkg/kubelet/kubelet.go

```diff
@@ -1811,7 +1811,7 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 	} else {
 		addr := net.ParseIP(kl.hostname)
 		if addr != nil {
-			node.Status.Addresses = []api.NodeAddress{{Type: api.NodeLegacyHostIP, Address: addr.String()}}
+			node.Status.Addresses = []api.NodeAddress{{Type: api.NodeInternalIP, Address: addr.String()}}
```

Comment:
I believe this comes from --hostname-override. Unfortunately, we have no way of knowing whether that's internal or external. I think internal is a reasonable guess. It's only used in the case where there isn't a cloud provider. A comment here would be useful.

@dchen1107 @cjcullen